yt_dlp/extractor/tiktok.py

```diff
@@ -263,7 +263,7 @@ def extract_addr(addr, add_meta={}):
 
         return {
             'id': aweme_id,
-            'title': aweme_detail['desc'],
+            'title': aweme_detail.get('desc'),
```

Comment:
Isn't `title` still a required field? You could use:
```py
            'title': aweme_detail.get('desc') or self._generic_title(url)
```

If appropriate, override the `_generic_title()` method in the extractor. 

Whereas the next line should use `get()`:
```py
            'description': aweme_detail.get('desc'),
```

Reply:
title is mandatory in the sense that extractors must return it. But returning None as title is allowed in yt-dlp. In this case, `YoutubeDL` will construct a generic title https://github.com/yt-dlp/yt-dlp/blob/5fa3c9a88f597625296981a4a26be723e65d4842/CONTRIBUTING.md?plain=1#L261

```
❯ yt-dlp -O title= https://www.tiktok.com/@pokemonlife22/video/7059698374567611694
WARNING: Extractor failed to obtain "title". Creating a generic title instead
title = 'TikTok video #7059698374567611694'
```

Reply:
Although I've always been somewhat sceptical about yt-dl's mandatory `title`, I can't help wondering why it's important that `info_dict['title']` should work even if `info_dict.get('title') is None`.

Also s/it's/its/ in https://github.com/yt-dlp/yt-dlp/blob/5fa3c9a88f597625296981a4a26be723e65d4842/CONTRIBUTING.md?plain=1#L261!

Reply:
I believe this decision was originally made because the filename is derived from title (before `-o` was a thing). If you are asking why I decided to make yt-dlp generate a title instead of making it optional; the main reason is to maintain compatibility with youtube-dl. As a bonus, this ensures that the default output template is somewhat sensible even when title extraction fails. I got the idea of `{extractor} video #{id}` from the facebook extractor

Reply:
All understandable, but why not generate the title for the `'title' not in info_dict` case too?

Reply:
Because I can't think of a good reason why an extractor shouldn't at least try to return a proper title. I wanted to make make downloads work even when title extraction fails, but keep it as a mandatory field that the extractor should always return

Reply:
Another reason: an extractor doesn't know if it's being invoked via `url_transparent` with a valid `title` already set from the invoking page.